### PR TITLE
Wrong release version for EVP_DigestSqueeze() in documentation

### LINKED ISSUE #23461: Wrong release version for EVP_DigestSqueeze() in documentation
https://github.com/openssl/openssl/blob/ed0f79c7ae63f7f29c9bfce2e0f960f0803be350/doc/man3/EVP_DigestInit.pod?plain=1#L808-L809

This is not correct, `EVP_DigestSqueeze()` has not yet been in any release.

### PR DESCRIPTION
**Pull request by shahsb**
The mentioned functions are targeted for 3.3.0

Fixes #23461

CLA: trivial

##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: doc/man3/EVP_DigestInit.pod [hunk 1]
```diff
@@ -806,7 +806,7 @@ EVP_MD_CTX_update_fn() and EVP_MD_CTX_set_update_fn() were deprecated
 in OpenSSL 3.0.
 
 The functions EVP_MD_CTX_dup() and EVP_DigestSqueeze() were added in
-OpenSSL 3.2.
+OpenSSL 3.3.0.
```
Comment (mattcaswell) [inline doc/man3/EVP_DigestInit.pod:809]: Probably the patch version is not necessary here, i.e. just keep it to 3.3

**Review by mattcaswell — CHANGES_REQUESTED** [review]

File: doc/man3/EVP_DigestInit.pod (hunk 1 above)
Comment (t8m) [reply doc/man3/EVP_DigestInit.pod:809]: Also EVP_MD_CTX_dup() was added in OpenSSL 3.1 - so this needs to be split.

Comment (shahsb) [reply doc/man3/EVP_DigestInit.pod]: Done. Addressed it in latest commit.

**Review by mattcaswell — APPROVED** [review]

**Comment by mattcaswell** [comment]
Agree trivial.

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Merged to the master branch. Thank you.

If you would like to submit a PR against the 3.2 branch that corrects the version when EVP_MD_CTX_dup() was introduced, it would be welcome.

**Pull request closed without merging**